apps/src/templates/certificates/SocialShare.jsx

```diff
@@ -93,7 +103,7 @@ export default function SocialShare({
           </button>
         </a>
       )}
-      <a href={print} className="social-print-link">
+      <a href={print} className="social-print-link" onClick={onPrint}>
```

Comment:
Since attaching an Amplitude logging event to a link can be inconsistent with it sometimes not logging fast enough, would it be possible to have the print event trigger when the user visits the print page (I think in [this file](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/sites/studio/pages/print_certificates/batch.js))? Then you also wouldn't need an `onClick` in an `a` tag.

Reply:
Hm, I think this link actually uses [this HAML file](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/views/print_certificates/show.html.haml), which doesn't run any JS, so I'd have to add that in. I believe the function to log to Amplitude will run before the navigation happens, but I think you're right that there's a race condition here. 

@dmcavoy thoughts on whether it would be better to possibly lose events on the button click or to log when someone visits the page to print a certificate?

Reply:
oh that does make it easier! I want to wait for Dani to weigh in though as there is a difference between the page view and the click event

Reply:
If the button would always open the page then I think it makes sense to just do the page view. Is there another way they would view the page?

Reply:
The only other way I know of after a quick search is the path in this video. 

https://github.com/code-dot-org/code-dot-org/assets/46464143/6fc2bb82-5b6c-4bdf-9b92-8a582bb4cc5e